sdk/messaging/azservicebus/client.go

```diff
@@ -369,3 +384,18 @@ func (client *Client) getCleanupForCloseable() (uint64, func()) {
 		client.linksMu.Unlock()
 	}
 }
+
+// getHostName returns fullyQualifiedNamespace if it is set, otherwise it returns the host name from the connection string.
+// If the connection string is not in the expected format, it returns an empty string.
+func getHostName(creds clientCreds) string {
+	if creds.fullyQualifiedNamespace != "" {
+		return creds.fullyQualifiedNamespace
+	}
+
+	parts := strings.Split(creds.connectionString, "/")
```

Comment:
We have an actual parser for the connection string, we should definitely use that. Look in `internal/conn`.